Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm

```diff
@@ -265,34 +275,53 @@ - (UIImage *)previewController:(QLPreviewController *)controller transitionImage
     [m_qlPreviewController setDataSource:m_qlPreviewControllerDataSource.get()];
 
     [presentingViewController presentViewController:m_qlPreviewController.get() animated:YES completion:nullptr];
+#endif
+}
+
+void SystemPreviewController::setDestinationURL(URL url)
+{
+    m_destinationURL = url;
 }
 
 void SystemPreviewController::updateProgress(float progress)
 {
+#if HAVE(UIKIT_WEBKIT_INTERNALS)
+    UNUSED_PARAM(progress);
+#else
     if (m_qlPreviewControllerDataSource)
         [m_qlPreviewControllerDataSource setProgress:progress];
+#endif
 }
 
 void SystemPreviewController::finish(URL url)
 {
+#if HAVE(UIKIT_WEBKIT_INTERNALS)
+    NSURL *nsurl = (NSURL*)url;
```

Comment:
There should be a space before the `*` in the cast. (Looks like there are two instances of that being incorrect in this file in `previewController:previewItemAtIndex:` and `finish:` If you want to fix them too.